### PR TITLE
Checkable Term Stats

### PR DESCRIPTION
**Pull request by sadasant**
WIP

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  The changelog has not been updated. Please update the changelog.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  The version was not updated. Please update the version.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  The README has not been updated. Please update the README.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  Could not find any browser results.
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>

  We were able to automatically fix some formatting issues in this PR for you!
  </td>
    </tr>
  </tbody>
</table>


### Some things that were possibly fixed:

- Code that could be fixed via the --fix flag
- Formatting that could be fixed by prettier

Take a look at this commit to see what happened in detail: f6b7400e2b53360effebef7d4d039bba19765527


And look at this wiki page to see the reasoning behind the ESLint rules: https://github.com/smartprocure/eslint-config-smartprocure/wiki/Rules-and-Why-We-Chose-Them
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/exampleTypes/TermsStatsTable.js [hunk 1]
```diff
@@ -29,7 +33,24 @@ let TermsStatsTable = injectTreeNode(
           {...{
             ...props,
             children: criteria
-              ? [
+              ? _.compact([
+                  checkable ? (
```
Comment (daedalus28) [inline src/exampleTypes/TermsStatsTable.js]: Lets make a CheckableTermsStatsTable like we did for results so this component stays simpler

File: src/exampleTypes/CheckableTermsStatsTable.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import _ from 'lodash/fp'
+import F from 'futil-js'
+import React from 'react'
+import { observer } from 'mobx-react'
+import { Column } from '../layout/ExpandableTable'
+import TermsStatsTable from './TermsStatsTable'
+
+let CheckableTermsStatsTable = observer(
+  ({
+    criteria,
```
Comment (daedalus28) [inline src/exampleTypes/CheckableTermsStatsTable.js]: by destructuring this up here, it means `criteria` isn't in `props` and so isn't passed to the underlying `TermsStatsTable`

File: src/exampleTypes/CheckableTermsStatsTable.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    children,
+    Checkbox,
+    checkable,
```
Comment (daedalus28) [inline src/exampleTypes/CheckableTermsStatsTable.js]: this doesn't seem used anymore

File: stories/imdb/stories/filterList.js [hunk 4]
```diff
@@ -109,108 +110,124 @@ let schemas = fromPromise(
...
+                  <DateHistogram
+                    path={['searchRoot', 'releases']}
+                    format={formatYear}
+                  />
+                  <CheckableTermsStatsTable
+                    criteria={['searchRoot', 'criteria']}
+                    path={['searchRoot', 'genreScores']}
+                    tableAttrs={{ style: { margin: 'auto' } }}
+                    Checkbox={() => <input type="checkbox" />}
+                    selected={selected}
```
Comment (daedalus28) [inline stories/imdb/stories/filterList.js]: does this work? doesn't checkboxValues expect a lens?

File: src/exampleTypes/CheckableTermsStatsTable.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import { Column } from '../layout/ExpandableTable'
+import TermsStatsTable from './TermsStatsTable'
+
+let CheckableTermsStatsTable = observer(
+  ({ children, Checkbox, getValue, selected, ...props }) => (
+    <TermsStatsTable
+      {...{
+        ...props,
+        children: [
+          <Column
```
Comment (dshishkov) [inline src/exampleTypes/CheckableTermsStatsTable.js]: If the intent is to have a checkbox on each row, I don't think this needs to use the Column's expand object, you can just move the display function as a prop on the Column. Unless I am missing something on how it should work?

File: stories/imdb/stories/filterList.js [hunk 6]
```diff
@@ -109,108 +110,124 @@ let schemas = fromPromise(
...
-                    label=""
-                    expand={{
-                      display: x =>
-                        `Show top 50 based on meta score for ${x} ▼`,
-                    }}
-                    collapse={{
-                      display: x =>
-                        `Hide top 50 based on meta score for ${x} ▲`,
-                    }}
+const Story = inject(() => {
```
Comment (daedalus28) [inline stories/imdb/stories/filterList.js]: instead of indenting this in an inject, move this + observer to the export

Comment (sadasant) [reply stories/imdb/stories/filterList.js]: this actually breaks the story :0

**Pull request merged**